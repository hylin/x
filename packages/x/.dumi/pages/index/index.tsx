import { createStyles } from 'antd-style';
import classnames from 'classnames';
import React from 'react';
import CompIntroduction from './components/CompIntroduction';
import DesignBanner from './components/DesignBanner';
import DesignFramework from './components/DesignFramework';
import DesignGuide from './components/DesignGuide';
import MainBanner from './components/MainBanner';
import SceneIntroduction from './components/SceneIntroduction';

const useStyle = createStyles(({ token, css }) => {
  return {
    section: css`
      background: linear-gradient(180deg, #1e2226e6 0%, #1c2024 38%, #16191c 100%);
      border-radius: 40px 40px 0 0;
      backdrop-filter: blur(40px);
      display: flex;
      flex-direction: column;
      gap: ${token.pcContainerMargin}px;
      padding: ${token.pcContainerMargin}px 0;
    `,
    container: css`
      margin-top: -40px;
    `,
    framework: css`
      border-radius: 0;
      background-image: linear-gradient(90deg, #5a37e6 0%, #0059c9 100%);
    `,
  };
});

const Homepage: React.FC = () => {
  const { styles } = useStyle();

  return (
    <main>
      <MainBanner />
      <section className={styles.section}>
        <DesignBanner />
      </section>
      <section className={classnames(styles.section, styles.container)}>
        <DesignGuide />
      </section>
      <section className={classnames(styles.section, styles.container)}>
        <SceneIntroduction />
        <CompIntroduction />
      </section>
      <section className={classnames(styles.section, styles.framework, styles.container)}>
        <DesignFramework />
      </section>
    </main>
  );
};

export default Homepage;
