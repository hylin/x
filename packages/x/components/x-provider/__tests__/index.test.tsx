import React from 'react';
import { Bubble } from '../../index';
import XProvider from '../index';

import mountTest from '../../../tests/shared/mountTest';
import rtlTest from '../../../tests/shared/rtlTest';
import themeTest from '../../../tests/shared/themeTest';
import { render } from '../../../tests/utils';

import type { XProviderProps } from '../index';

const xProviderProps: XProviderProps = {
  bubble: {
    className: 'test-bubble',
  },
  conversations: {
    className: 'test-conversations',
  },
  prompts: {
    className: 'test-prompts',
  },
  sender: {
    className: 'test-sender',
  },
  suggestion: {
    className: 'test-suggestion',
  },
  thoughtChain: {
    className: 'test-thoughtChain',
  },
};

describe('XProvider Component', () => {
  mountTest(() => <XProvider {...xProviderProps} />);

  rtlTest(() => <XProvider {...xProviderProps} />);

  themeTest(() => <XProvider {...xProviderProps} />);

  beforeAll(() => {
    jest.useFakeTimers();
  });

  afterAll(() => {
    jest.useRealTimers();
  });

  it('bubble.className', () => {
    const { container } = render(
      <XProvider {...xProviderProps}>
        <Bubble content="test" />
      </XProvider>,
    );
    const element = container.querySelector('.test-bubble');
    expect(element).toBeTruthy();
  });
});
