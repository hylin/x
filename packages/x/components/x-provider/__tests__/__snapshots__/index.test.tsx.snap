// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`XProvider Component rtl render component should be rendered correctly in RTL direction 1`] = `null`;

exports[`XProvider Component test theme component should be rendered correctly when configuring the theme.components 1`] = `null`;
